DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Rules/AnonymizerRuleFactory.cs

```diff
@@ -0,0 +1,136 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.AnonymizerConfigurations;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Rules
+{
+    public class AnonymizerRuleFactory : IAnonymizerRuleFactory
+    {
+        private readonly AnonymizerDefaultSettings _defaultSettings;
+
+        private readonly Dictionary<string, JObject> _customizedSettings;
+
+        private readonly IAnonymizerProcessorFactory _processorFactory;
+
+        private readonly IAnonymizerSettingsFactory _settingsFactory;
+
+        public AnonymizerRuleFactory(AnonymizerConfiguration configuration, IAnonymizerProcessorFactory processorFactory = null, IAnonymizerSettingsFactory settingsFactory = null)
+        {
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+
+            _defaultSettings = configuration.DefaultSettings;
+            _customizedSettings = configuration.CustomizedSettings;
+            _processorFactory = processorFactory;
+            _settingsFactory = settingsFactory;
+        }
+
+        public AnonymizerRule[] CreateAnonymizationDicomRule(JObject[] ruleContent)
```

Comment:
>ruleContent [](http://example.com/codeflow?start=71&length=11)

ruleContents might be better since it's an array of objects.

Reply:
Thanks, fixed it.